gencred/cmd/gencred/main.go

```diff
@@ -60,7 +61,7 @@ func (o *options) parseFlags() {
 	flag.StringVarP(&o.output, "output", "o", defaultConfigFileName, "Output path for generated kubeconfig file.")
 	flag.BoolVarP(&o.certificate, "certificate", "c", false, "Authorize with a client certificate and key.")
 	flag.BoolVarP(&o.serviceaccount, "serviceaccount", "s", false, "Authorize with a service account.")
-	flag.IntVar(&o.duration, "duration", defaultDurationInDays, "How many days the cred is valid, default is 7.")
+	flag.Int64Var(&o.duration, "duration", defaultDurationInDays, "How many seconds the cred is valid, default is 172800, can only set to be lower than the default.")
```

Comment:
This change is backwards incompatible. The flag was only introduced 20 days ago so it might be ok if we're confident we're the only people using it.